parity/stratum.rs

```diff
@@ -16,42 +16,97 @@
 
 //! Parity sync service
 
-use std;
 use std::sync::Arc;
+use std::sync::atomic::{AtomicBool, Ordering};
 use ethcore_stratum::{Stratum as StratumServer, PushWorkHandler, RemoteJobDispatcher, ServiceConfiguration};
-use std::thread;
 use modules::service_urls;
 use boot;
 use hypervisor::service::IpcModuleId;
-use std::net::SocketAddr;
+use hypervisor::{HYPERVISOR_IPC_URL, ControlService};
+use std::net::{SocketAddr, IpAddr};
 use std::str::FromStr;
+use nanoipc;
+use std::thread;
+use ethcore::miner::stratum::{STRATUM_SOCKET_NAME, JOB_DISPATCHER_SOCKET_NAME};
+
+pub const MODULE_ID: IpcModuleId = 8000;
 
-const STRATUM_MODULE_ID: IpcModuleId = 8000;
+#[derive(Default)]
+struct StratumControlService {
+	pub stop: Arc<AtomicBool>,
+}
+
+impl ControlService for StratumControlService {
+	fn shutdown(&self) -> bool {
+		trace!(target: "hypervisor", "Received shutdown from control service");
+		self.stop.store(true, ::std::sync::atomic::Ordering::Relaxed);
+		true
+	}
+}
 
 pub fn main() {
 	boot::setup_cli_logger("stratum");
 
 	let service_config: ServiceConfiguration = boot::payload()
 		.unwrap_or_else(|e| panic!("Fatal: error reading boot arguments ({:?})", e));
```

Comment:
IMHO would be better to split `fn main()`  into `fn start() -> Result<..., String>` and avoid explicit panics (or just have a single place with explicit `panic` if it can be used as a separate binary.